core/examples/fs_module_loader.rs

```diff
@@ -1,12 +1,87 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
+use anyhow::anyhow;
 use anyhow::Context;
 use deno_core::anyhow::Error;
+use deno_core::v8;
+use deno_core::FastString;
 use deno_core::FsModuleLoader;
 use deno_core::JsRuntime;
+use deno_core::ModuleSourceCode;
 use deno_core::RuntimeOptions;
+use std::borrow::Cow;
+use std::collections::HashMap;
 use std::rc::Rc;
 
+fn custom_module_evaluation_cb(
+  scope: &mut v8::HandleScope,
+  module_type: Cow<'_, str>,
+  module_name: &FastString,
+  code: ModuleSourceCode,
+) -> Result<v8::Global<v8::Value>, Error> {
+  match &*module_type {
+    "bytes" => Ok(bytes_module(scope, code)),
+    "text" => text_module(scope, module_name, code),
+    "url" => Ok(url_module(scope, module_name)),
+    _ => Err(anyhow!(
+      "Can't import {:?} because of unknown module type {}",
+      module_name,
+      module_type
+    )),
+  }
+}
+
+fn bytes_module(
+  scope: &mut v8::HandleScope,
+  code: ModuleSourceCode,
+) -> v8::Global<v8::Value> {
+  // FsModuleLoader always returns bytes.
+  let ModuleSourceCode::Bytes(buf) = code else {
+    unreachable!()
+  };
+  let buf_len: usize = buf.len();
+  let backing_store = v8::ArrayBuffer::new_backing_store_from_vec(buf);
+  let backing_store_shared = backing_store.make_shared();
+  let ab = v8::ArrayBuffer::with_backing_store(scope, &backing_store_shared);
+  let uint8_array = v8::Uint8Array::new(scope, ab, 0, buf_len).unwrap();
+  let value: v8::Local<v8::Value> = uint8_array.into();
+  v8::Global::new(scope, value)
+}
+
+fn text_module(
+  scope: &mut v8::HandleScope,
+  module_name: &FastString,
+  code: ModuleSourceCode,
+) -> Result<v8::Global<v8::Value>, Error> {
+  // FsModuleLoader always returns bytes.
+  let ModuleSourceCode::Bytes(buf) = code else {
+    unreachable!()
+  };
+
+  let code = String::from_utf8(buf).with_context(|| {
+    format!("Can't convert {:?} source code to string", module_name)
```

Comment:
nit: `std::str::from_utf8` and pass the borrowed value to `v8::String::new`

Reply:
Thanks, applied. I'm still debating what should be the signature of `ModuleSourceCode::Bytes`. Given that it will be stored in the `deno_graph`'s graph it should probably be `Arc<[u8]>`. I'll discuss this with @dsherret.